nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EncryptAttributes.java

```diff
@@ -0,0 +1,611 @@
+            }
+            if (!CipherUtility.isValidKeyLengthForAlgorithm(keyBytes.length * 8, encryptionMethod.getAlgorithm())) {
+                List<Integer> validKeyLengths = CipherUtility.getValidKeyLengthsForAlgorithm(encryptionMethod.getAlgorithm());
+                validationResults.add(new ValidationResult.Builder().subject(RAW_KEY_HEX.getName())
+                        .explanation("Key must be valid length [" + StringUtils.join(validKeyLengths, ", ") + "]. See Admin Guide.").build());
+            }
+        }
+
+        // Perform some analysis on the selected encryption algorithm to ensure the JVM can support it and the associated key
+
+        List<String> kdfsForKeyedCipher = getKDFsForKeyedCipher();
+        if (kdf == null || !kdfsForKeyedCipher.contains(kdf.name())) {
+            validationResults.add(new ValidationResult.Builder().subject(KEY_DERIVATION_FUNCTION.getName())
+                    .explanation(KEY_DERIVATION_FUNCTION.getDisplayName() + " is required to be " + StringUtils.join(kdfsForKeyedCipher, ", ") + " when using algorithm " +
+                            encryptionMethod.getAlgorithm()).build());
+        }
+
+        return validationResults;
+    }
+
+    private List<String> getKDFsForKeyedCipher() {
+        List<String> kdfsForKeyedCipher = new ArrayList<>();
+        kdfsForKeyedCipher.add(KeyDerivationFunction.NONE.name());
+        for (KeyDerivationFunction k : KeyDerivationFunction.values()) {
+            if (k.isStrongKDF()) {
+                kdfsForKeyedCipher.add(k.name());
+            }
+        }
+        return kdfsForKeyedCipher;
+    }
+
+    private List<String> getKDFsForPBECipher(EncryptionMethod encryptionMethod) {
+        List<String> kdfsForPBECipher = new ArrayList<>();
+        for (KeyDerivationFunction k : KeyDerivationFunction.values()) {
+            // Add all weak (legacy) KDFs except NONE
+            if (!k.isStrongKDF() && !k.equals(KeyDerivationFunction.NONE)) {
+                kdfsForPBECipher.add(k.name());
+                // If this algorithm supports strong KDFs, add them as well
+            } else if ((encryptionMethod.isCompatibleWithStrongKDFs() && k.isStrongKDF())) {
+                kdfsForPBECipher.add(k.name());
+            }
+        }
+        return kdfsForPBECipher;
+    }
+
+    private Map<String, String> buildNewAttributes(FlowFile file, String atrList,
+                                                   EncryptContent.Encryptor encryptor,
+                                                   boolean isToBeEncrypted) throws Exception {
+
+        Map<String, String> oldAttributes = file.getAttributes();
+        Map<String, String> atrToWrite = new HashMap<>();
+        final String filenameAttr = CoreAttributes.FILENAME.key();
+        final String uuidAttr = CoreAttributes.UUID.key();
+
+        if (!StringUtils.isEmpty(atrList)) {
```

Comment:
I think we should log `INFO`, if set doesn't have any attributes "to-be-operated-on" and not for the empty set as this is clear from documentation.